internal/aks/create.go

```diff
@@ -128,6 +130,15 @@ func CreateOrUpdateCluster(ctx context.Context, cred *Credentials, clusterClient
 		}
 	}
 
+	addonProfiles := map[string]*containerservice.ManagedClusterAddonProfile{
+		"httpApplicationRouting": {
+			Enabled: spec.HTTPApplicationRouting,
+		},
+	}
+	if !hasHTTPApplicationRoutingSupport(spec) {
+		delete(addonProfiles, "httpApplicationRouting")
```

Comment:
Why delete after it's added? Why not
```suggestion
	if hasHTTPApplicationRoutingSupport(spec) {
		addonProfiles := map[string]*containerservice.ManagedClusterAddonProfile{
			"httpApplicationRouting": {
				Enabled: spec.HTTPApplicationRouting,
			},
		}
```

Reply:
good point